app/src/main/kotlin/com/exallium/todoapp/database/StubDatabaseImpl.kt

```diff
@@ -33,7 +33,11 @@ class StubDatabaseImpl(private val idFactory: IdFactory) : Database {
     }
 
     override fun deleteNote(note: Note): Single<Unit> {
-        db.remove(note.id)
-        return Single.just(Unit)
+        return if (!db.containsKey(note.id)) {
+            Single.error(DatabaseException("Note does not exist"))
+        } else {
+            db.remove(note.id)
+            Single.just(Unit)
+        }
```

Comment:
db.remove will return the removed note or null, so this can be more concisely expressed as

```
return if (db.remove(note.id) == null) {
    Single.error
} else {
    Single.just
}
```